google_guest_agent/sshca/sshca.go

```diff
@@ -63,7 +63,12 @@ func writeFile(ctx context.Context, evType string, data interface{}, evData *eve
 	}
 
 	// Make sure we close the pipe after we've done writing to it.
-	pipeData := evData.Data.(*sshtrustedca.PipeData)
+	pipeData, ok := evData.Data.(*sshtrustedca.PipeData)
+	if !ok {
+		logger.Errorf("Received invalid event data (%+v), ignoring this event and un-subscribing %s", evData.Data, evType)
```

Comment:
Is it possible for this to happen in normal operation? Should we ignore and resubscribe and just let the event structure tear down the callback on shutdown instead?

Reply:
No, since this is subscribing to specific event we always expect data of specific type and should never receive invalid data. If there was any error we'll see it in `evData.Error` in which case we do renew handler.